core/src/main/java/org/elasticsearch/search/suggest/Suggest.java

```diff
@@ -399,35 +400,17 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
                         "Cannot parse suggestion response without type information. Set [" + RestSearchAction.TYPED_KEYS_PARAM
                                 + "] parameter on the request to ensure the type information is added to the response output");
             }
-            Suggestion suggestion = null;
-            Function<XContentParser, Entry> entryParser = null;
-            // the "size" parameter and the SortBy for TermSuggestion cannot be parsed from the response, use default values
-            // TODO investigate if we can use NamedXContentRegistry instead of this switch
-            switch (type) {
-                case Suggestion.NAME:
```

Comment:
Okay, I was only suggesting that if we don't support it on the parsing side, we might as well remove it everywhere (it's use in CustomSuggester, making Suggestion class abstract, removing from transport layer etc...) and then it would probably be better to do this in a separate PR. I'm okay with not supporting the parser.